llvm/tools/obj2yaml/xcoff2yaml.cpp

```diff
@@ -146,10 +146,11 @@ Error XCOFFDumper::dumpFileAuxSym(XCOFFYAML::Symbol &Sym,
   return Error::success();
 }
 
-void XCOFFDumper::dumpStatAuxSym(XCOFFYAML::Symbol &Sym,
-                                 const XCOFFSymbolRef &SymbolEntRef) {
-  assert(Sym.NumberOfAuxEntries == 1 &&
-         "expected a single aux symbol for C_STAT!");
+Error XCOFFDumper::dumpStatAuxSym(XCOFFYAML::Symbol &Sym,
+                                  const XCOFFSymbolRef &SymbolEntRef) {
+  if (Sym.NumberOfAuxEntries != 1)
+    return createError("expected a single aux symbol for C_STAT, while got: " +
```

Comment:
Suggest that we  provide more detail info for invalid Symbol (for example symbol index), let user can locate the specific invalid symbol.    

and some test case for the Error scenarion.